test/testuser.py

```diff
@@ -6,18 +6,59 @@ def __init__(self, username, userapi):
         self._userapi = userapi
         self.username = username
 
+    def when_searching_with(self, queryParams):
+        return RecordingQueryPromise(self, queryParams)
 
-    def can_see_recording_from(self, testdevice):
-        recordings = self._userapi.query(limit=1)
+    def when_searching_with_tagmode(self, tagmode):
+        queryParams = {"tagmode": tagmode}
+        return RecordingQueryPromise(self, queryParams)
+
+    def when_searching_for_tags(self, *tags):
+        queryParams = {"tags": tags}
+        return RecordingQueryPromise(self, queryParams)
+
+    def can_see_recordings(self, *expectedTestRecordings):
```

Comment:
The convention in python is to use lower_case_with_underscores for variables names, not camelCase.